src/mpsc/blocking.rs

```diff
@@ -574,6 +575,80 @@ feature! {
             Some(recycling::take(&mut *val, self.recycle))
         }
 
+        /// Receives the next message for this receiver, **by reference**, waiting for at most `timeout`.
+        /// Returns an error if the corresponding channel has hung up, or if it waits more than `timeout`.
+        ///
```

Comment:
nit, take it or leave it: I might consider making this a bit closer to the  documentation for other methods: 
```suggestion
        ///
        /// If there are no messages in the channel's buffer, but the channel has
        /// not yet been closed, this method will block until a message is sent,
        /// the channel is closed, or the provided `timeout` has elapsed.
        ///
        /// # Returns
        ///
        /// - [`Ok`]`(`[`RecvRef`]`<T>)` if a message was received.
        /// - [`Err`]`(`[`RecvTimeoutError::Timeout`]`)` if the timeout has elapsed.
        /// - [`Err`]`(`[`RecvTimeoutError::Closed`]`)` if the channel has closed.
```